src/collector/tweak_score_top_collector.rs

```diff
@@ -121,9 +121,112 @@ where
 
 impl<F, TScore> ScoreSegmentTweaker<TScore> for F
 where
-    F: 'static + Sync + Send + Fn(DocId, Score) -> TScore,
+    F: 'static + Sync + Send + FnMut(DocId, Score) -> TScore,
 {
-    fn score(&self, doc: DocId, score: Score) -> TScore {
+    fn score(&mut self, doc: DocId, score: Score) -> TScore {
         (self)(doc, score)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::collector::TopDocs;
+    use crate::query::BooleanQuery;
+    use crate::schema::*;
+    use crate::{DocId, Index, Score, SegmentReader};
+    use std::collections::HashSet;
+
+    #[test]
+    fn tweak_score_with_facets() {
```

Comment:
This looks more like an example than a test. Could you move this to an example?

Reply:
Absolutely! Great idea, I knew it felt wrong here.